src/operations/aggregate.ts

```diff
@@ -65,7 +64,7 @@ export class AggregateOperation<T = Document> extends CommandOperation<T> {
     }
 
     if (this.hasWriteStage) {
-      this.readPreference = ReadPreference.primary;
+      this.trySecondaryWrite = true;
```

Comment:
We now flag the operation to try executing the write on a secondary instead of setting a primary read preference. This is not specific to $out/$merge as I wanted a bit more generic name in case more appear in the future.